src/plugins/intel_cpu/src/nodes/interaction.cpp

```diff
@@ -0,0 +1,228 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <algorithm>
+#include <chrono>
+#include <cmath>
+#include <string>
+#include <vector>
+
+#include "ie_parallel.hpp"
+#include "ngraph_transformations/op/interaction.hpp"
+#include "interaction.h"
+#include "utils/general_utils.h"
+#include <onednn/dnnl.h>
+#include <dnnl_extension_utils.h>
+#include "nodes/common/cpu_convert.h"
+#include "memory_desc/cpu_memory_desc_utils.h"
+#include "memory_desc/dnnl_blocked_memory_desc.h"
+#include "common/bfloat16.hpp"
+#include "common/cpu_memcpy.h"
+#include <cpu/x64/cpu_isa_traits.hpp>
+
+namespace ov {
+namespace intel_cpu {
+namespace node {
+
+Interaction::Interaction(const std::shared_ptr<ngraph::Node>& op, const dnnl::engine& eng, WeightsSharing::Ptr &cache)
+        : Node(op, eng, cache) {
+    std::string errorMessage;
+    if (isSupportedOperation(op, errorMessage)) {
+        errorPrefix = "Interaction node with name '" + getName() + "'";
+    } else {
+        IE_THROW(NotImplemented) << errorMessage;
+    }
+}
+
+void Interaction::initSupportedPrimitiveDescriptors() {
+    if (!supportedPrimitiveDescriptors.empty())
+        return;
+    dataPrecision = getOriginalInputPrecisionAtPort(0);
+    // Current impl only support FP32 BF16, BF16 is preferred
+    if (dataPrecision != InferenceEngine::Precision::FP32 && dnnl::impl::cpu::x64::mayiuse(dnnl::impl::cpu::x64::avx512_core_bf16)) {
+        dataPrecision = InferenceEngine::Precision::BF16;
+    } else {
+        dataPrecision = InferenceEngine::Precision::FP32;
+    }
+    // initialize input ports
+    std::vector<PortConfigurator> inPortConfigs;
+    for (size_t i = 0; i < getParentEdges().size(); ++i) {
+        inPortConfigs.emplace_back(
+            LayoutType::ncsp,
+            dataPrecision,
+            getInputShapeAtPort(i),
+            false, -1);
+    }
+    // initialize output port
+    std::vector<PortConfigurator> outPortConfigs = {
+        PortConfigurator {
+            LayoutType::ncsp,
+            dataPrecision,
+            getOutputShapeAtPort(0),
+            false,
+            -1
+        }
+    };
+    //add descriptor
+    addSupportedPrimDesc(inPortConfigs, outPortConfigs, impl_desc_type::ref_any, true);
+}
+
+template <typename T>
```

Comment:
These functions performs data movement only and require just data type size, so they should not be templated. That allows to reduce binary size.